.github/workflows/publish.agent-wheel.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish Agent Wheel
+
+on:
+  workflow_dispatch: # manual trigger
+    inputs:
+      buildConfig:
+        description: Build Configuration
+        required: true
+        default: release
+  push:
+    branches: [main]
+
+jobs:
+  publish-wheel:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    env:
+      PYTHON_VERSION: "3.9"
+      POETRY_VERSION: "1.1.14"
+      WORKDIR: "./app/HutchAgent"
+      ARTIFACT_NAME: "hutchagent-dist"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install poetry
+        working-directory: ${{ env.WORKDIR }}
+        run: pip install poetry==${{ env.POETRY_VERSION }}
+      - name: Install agent deps
+        working-directory: ${{ env.WORKDIR }}
+        run: poetry install --no-dev
+      - name: Build dist
+        working-directory: ${{ env.WORKDIR }}
+        run: poetry build
+      - uses: ncipollo/release-action@v1.10.0
```

Comment:
I'm not sure we want this workflow to produce a GitHub release?

We should definitely upload an artifact on the workflow run, just like the Publish Manager workflow does though.

As for the releases, it depends on whether we intend to release versions of components individually, or the whole stack all at once. I guess that's a discussion we need to have.

Also I guess this current way doesn't work repeatedly, due to the static tag `hutchagent`? normally you'd want some version information in a release.

Reply:
I was wondering about the versioning bit too. I'll change it to build the artefacts for now and we can work out how releases work later

Reply:
yeah, there's another pbi (AB#69936) for delivery/deployment of these artifacts anyway, which I would consider this to come under.